lib/pghero/methods/space.rb

```diff
@@ -1,49 +1,147 @@
+                                         FROM (
+                                           SELECT 
+                                             pg_database_size(current_database()) AS size 
+                                           UNION ALL 
+                                           SELECT 
+                                             result::bigint AS size 
+                                           FROM run_command_on_workers($cmd$ SELECT 
+                                                                               pg_database_size(current_database()); 
+                                                                       $cmd$)) AS worker_size")
+        end
       end
 
       def relation_sizes
-        select_all_size <<-SQL
-          SELECT
-            n.nspname AS schema,
-            c.relname AS relation,
-            CASE WHEN c.relkind = 'r' THEN 'table' ELSE 'index' END AS type,
-            pg_table_size(c.oid) AS size_bytes
-          FROM
-            pg_class c
-          LEFT JOIN
-            pg_namespace n ON n.oid = c.relnamespace
-          WHERE
-            n.nspname NOT IN ('pg_catalog', 'information_schema')
-            AND n.nspname !~ '^pg_toast'
-            AND c.relkind IN ('r', 'i')
-          ORDER BY
-            pg_table_size(c.oid) DESC,
-            2 ASC
-        SQL
+        if !citus_enabled?
+          select_all_size <<-SQL
+            SELECT
+              n.nspname AS schema,
+              c.relname AS relation,
+              CASE WHEN c.relkind = 'r' THEN 'table' ELSE 'index' END AS type,
+              pg_table_size(c.oid) AS size_bytes
+            FROM
+              pg_class c
+            LEFT JOIN
+              pg_namespace n ON n.oid = c.relnamespace
+            WHERE
+              n.nspname NOT IN ('pg_catalog', 'information_schema')
+              AND n.nspname !~ '^pg_toast'
+              AND c.relkind IN ('r', 'i')
+            ORDER BY
+              pg_table_size(c.oid) DESC,
+              2 ASC
+          SQL
+        else
+          select_all_size <<-SQL
+            WITH dist_indexes_sizes AS ( 
+              WITH dist_indexes(idxname, tablename) AS ( 
+                SELECT 
+                  indexrelid, 
+                  indrelid 
+                FROM   
+                  pg_index 
+                JOIN   
+                  pg_dist_partition ON (indrelid = logicalrelid)
+              ), 
+              dist_indexes_shard_sizes AS ( 
+                SELECT 
+                  idxname::regclass, 
+                  (run_command_on_placements(tablename::regclass, $$
+                    SELECT 
+                      pg_table_size(replace('%s','$$ || tablename::regclass::text || $$', '$$ || idxname::regclass::text || $$')) 
+                  $$)).result
+                FROM   
+                  dist_indexes 
+              ) 
+              SELECT   
+                idxname, 
+                sum(result::bigint) AS idxsize
```

Comment:
`sum`s should be uppercase. As a rule of thumb most SQL keywords(except type, extension, UDF names) are uppercase, remaining things are lowercase.